odoo/addons/base/models/ir_attachment.py

```diff
@@ -88,6 +88,15 @@ def _get_path(self, bin_data, sha):
         dirname = os.path.dirname(full_path)
         if not os.path.isdir(dirname):
             os.makedirs(dirname)
+        # prevent sha-1 collision
+        if os.path.isfile(full_path):
+            stored_sha2 = self._compute_file_checksum(full_path, hashlib.sha512)
+            new_sha2 = hashlib.sha512(bin_data or b'').hexdigest()
+            if stored_sha2 != new_sha2:
```

Comment:
Why not comparing both contents byte-by-byte?  Hashing them is just much slower...
I don't see any added value in having their SHA-2.

Reply:
done